stan/math/prim/fun/grad_pFq.hpp

```diff
@@ -2,344 +2,201 @@
 #define STAN_MATH_PRIM_FUN_GRAD_PFQ_HPP
 
 #include <stan/math/prim/meta.hpp>
-#include <stan/math/prim/err.hpp>
-#include <stan/math/prim/fun/constants.hpp>
-#include <stan/math/prim/fun/exp.hpp>
-#include <stan/math/prim/fun/fabs.hpp>
-#include <stan/math/prim/fun/log.hpp>
-#include <stan/math/prim/fun/elt_multiply.hpp>
-#include <stan/math/prim/fun/log_rising_factorial.hpp>
 #include <stan/math/prim/fun/hypergeometric_pFq.hpp>
-#include <stan/math/prim/fun/max.hpp>
-#include <stan/math/prim/fun/log_sum_exp.hpp>
+#include <stan/math/prim/fun/add.hpp>
 #include <stan/math/prim/fun/prod.hpp>
-#include <stan/math/prim/fun/sign.hpp>
-#include <stan/math/prim/fun/transpose.hpp>
-#include <cmath>
-#include <iostream>
+#include <stan/math/prim/fun/log.hpp>
+#include <stan/math/prim/fun/log1p.hpp>
+#include <stan/math/prim/fun/abs.hpp>
+#include <stan/math/prim/fun/floor.hpp>
+#include <stan/math/prim/fun/max.hpp>
+#include <stan/math/prim/fun/select.hpp>
+#include <stan/math/prim/fun/sum.hpp>
+#include <stan/math/prim/fun/as_column_vector_or_scalar.hpp>
+#include <stan/math/prim/fun/value_of_rec.hpp>
 
 namespace stan {
 namespace math {
-namespace internal {
+
 /**
  * Returns the gradient of generalized hypergeometric function wrt to the
  * input arguments:
  * \f$ _pF_q(a_1,...,a_p;b_1,...,b_q;z) \f$
  *
- * The derivatives wrt a and b are defined using a Kampé de Fériet function,
- *   (https://en.wikipedia.org/wiki/Kamp%C3%A9_de_F%C3%A9riet_function).
- *   This is implemented below as an infinite sum (on the log scale) until
- *   convergence.
- *
- * \f$ \frac{\partial}{\partial a_1} =
- *     \frac{z\prod_{j=2}^pa_j}{\prod_{j=1}^qb_j}
- *     F_{q+1\:0\;1}^{p\:1\:2}\left(\begin{array}&a_1+1,...,a_p+1;1;1,a_1\\
- *        2, b_1+1,...,b_1+1;;a_1+1\end{array};z,z\right) \f$
+ * Where:
+ * \f$
+ * \frac{\partial }{\partial a_1} =
+ *  \sum_{k=0}^{\infty}{
+ *    \frac
+ *      {\psi\left(k+a_1\right)\left(\prod_{j=1}^p\left(a_j\right)_k\right)z^k}
+ *      {k!\prod_{j=1}^q\left(b_j\right)_k}}
+ *    - \psi\left(a_1\right)_pF_q(a_1,...,a_p;b_1,...,b_q;z)
```

Comment:
```suggestion
 *    - \psi\left(a_1\right){}_pF_q(a_1,...,a_p;b_1,...,b_q;z)
```

Reply:
The proposed change renders the math a little cleaner. 

The new is on top, the old is on bottom. 
<img width="253" alt="image" src="https://github.com/stan-dev/math/assets/425751/c7f7be2e-05d9-42de-8c62-c4feeec7fa31">

If you look carefully, the `p` is closer to F, rather than the `p` looking like a subscript to the `psi` function call.